python/graphstorm/model/node_gnn.py

```diff
@@ -187,18 +189,46 @@ def node_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
             input_feats = data.get_node_feats(input_nodes, device)
             blocks = [block.to(device) for block in blocks]
             pred, emb = model.predict(blocks, input_feats, None, input_nodes, return_proba)
-            preds.append(pred.cpu())
-            embs.append(emb.cpu())
+            for ntype, ntype_labels in data.get_labels(seeds).items():
+                if ntype in labels:
+                    labels[ntype].append(ntype_labels.cpu())
+                else:
+                    labels[ntype] = [ntype_labels.cpu()]
+            if isinstance(pred, dict):
+                for ntype, ntype_pred in pred.items():
+                    if ntype in preds:
+                        preds[ntype].append(ntype_pred.cpu())
+                    else:
+                        preds[ntype] = [ntype_pred.cpu()]
```

Comment:
i see this is replicated everywhere. do you mind creating a function for this operation.

def _add(list_dict, ntype, tensor):
    if ntype in list_dict:
        list_dict[ntype].append(tensor.cpu())
    else:
        list_dict[ntype] = [tensor.cpu()]

Reply:
Done. Please refer to please refer to function `append_to_dict(from_dict, to_dict):`